gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/testing/LocalResourceManagerHelper.java

```diff
@@ -0,0 +1,588 @@
+import com.google.common.io.ByteStreams;
+
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import org.joda.time.format.ISODateTimeFormat;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.HttpURLConnection;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.logging.Logger;
+import java.util.zip.GZIPInputStream;
+
+/**
+ * Utility to create a local Resource Manager mock for testing.
+ *
+ * The mock runs in a separate thread, listening to port 8080 for HTTP requests.
+ */
+@SuppressWarnings("restriction")
+public class LocalResourceManagerHelper {
+  private static final Logger log = Logger.getLogger(LocalResourceManagerHelper.class.getName());
+  private static final JsonFactory jsonFactory =
+      new com.google.api.client.json.jackson.JacksonFactory();
+  private static final Random PROJECT_NUMBER_GENERATOR = new Random();
+
+  // see https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects
+  private static final Set<Character> PERMISSIBLE_PROJECT_NAME_PUNCTUATION =
+      ImmutableSet.of('-', '\'', '"', ' ', '!');
+
+  private HttpServer server;
+  private final ConcurrentHashMap<String, Project> projects = new ConcurrentHashMap<>();
+
+  static class Response {
+    private final int code;
+    private final String body;
+
+    Response(int code, String body) {
+      this.code = code;
+      this.body = body;
+    }
+
+    int code() {
+      return code;
+    }
+
+    String body() {
+      return body;
+    }
+  }
+
+  enum Error {
+    ALREADY_EXISTS(
+        409, "global", "Requested entity already exists.", "alreadyExists", "ALREADY_EXISTS"),
+    PERMISSION_DENIED(
+        403, "global", "The caller does not have permission.", "forbidden", "PERMISSION_DENIED"),
+    FAILED_PRECONDITION( // change this error code to 412 when #440 is fixed
+        400, "global", "Precondition check failed.", "failedPrecondition", "FAILED_PRECONDITION"),
+    INVALID_ARGUMENT( // change this error code to 412 when #440 is fixed
+        400, "global", "Request contains an invalid argument.", "badRequest",
+        "INVALID_ARGUMENT");
+
+    private final Response response;
+
+    Error(int code, String domain, String message, String reason, String status) {
+      this.response = new Response(code, toJson(code, domain, message, reason, status));
+    }
+
+    private static String toJson(
+        int code, String domain, String message, String reason, String status) {
+      Map<String, Object> args = new HashMap<>();
+      Map<String, Object> nestedArgs = new HashMap<>();
```

Comment:
why not call this variable errors and put it directly as `args.put("errors", ImmutableList.of(errors))`